src/poetry/utils/env/base_env.py

```diff
@@ -348,7 +348,7 @@ def _run(self, cmd: list[str], **kwargs: Any) -> str:
                 output = ""
             else:
                 output = subprocess.check_output(
-                    cmd, stderr=stderr, env=env, text=True, **kwargs
+                    cmd, stderr=stderr, env=env, text=True, encoding="utf-8", **kwargs
```

Comment:
this one looks suspect - why should we assume that the subprocess is producing utf8 output?

looks like callers to this arrive via `run_python_scipt()`, which so far as I can see does not do anything to force utf8

Reply:
Hi @dimbleby, thanks for the feedback! I think you are right, I updated it so that it enforces the locale encoding.